x-pack/plugins/infra/public/components/asset_details/content/content.tsx

```diff
@@ -47,7 +53,24 @@ export const Content = () => {
           <Processes />
         </TabPanel>
         <TabPanel activeWhen={ContentTabIds.PROFILING}>
-          <Profiling />
+          {loading ? (
+            <div
+              css={css`
+                display: flex;
+                justify-content: center;
+              `}
+            >
+              <EuiLoadingSpinner size="m" />
+            </div>
+          ) : (
+            <>
+              {error !== null && response?.has_setup === false ? (
+                <ProfilingEmptyState />
+              ) : (
+                <Profiling />
+              )}
+            </>
+          )}
```

Comment:
I moved that to the Profiling component, I had implemented it here mainly because of the DatePicker, but as you suggested I moved that to the Profiling component too, so no need to leave it here. Thanks